include/zephyr/posix/unistd.h

```diff
@@ -55,6 +251,137 @@ pid_t getpid(void);
+#define __z_posix_sysconf_SC_2_PBS_ACCOUNTING		  _POSIX2_PBS_ACCOUNTING
+#define __z_posix_sysconf_SC_2_PBS_CHECKPOINT		  _POSIX2_PBS_CHECKPOINT
+#define __z_posix_sysconf_SC_2_PBS_LOCATE		  _POSIX2_PBS_LOCATE
+#define __z_posix_sysconf_SC_2_PBS_MESSAGE		  _POSIX2_PBS_MESSAGE
+#define __z_posix_sysconf_SC_2_PBS_TRACK		  _POSIX2_PBS_TRACK
+#define __z_posix_sysconf_SC_2_SW_DEV			  _POSIX2_SW_DEV
+#define __z_posix_sysconf_SC_2_UPE			  _POSIX2_UPE
+#define __z_posix_sysconf_SC_2_VERSION			  _POSIX2_VERSION
+#define __z_posix_sysconf_SC_XOPEN_CRYPT		  _XOPEN_CRYPT
+#define __z_posix_sysconf_SC_XOPEN_ENH_I18N		  _XOPEN_ENH_I18N
+#define __z_posix_sysconf_SC_XOPEN_REALTIME		  _XOPEN_REALTIME
+#define __z_posix_sysconf_SC_XOPEN_REALTIME_THREADS	  _XOPEN_REALTIME_THREADS
+#define __z_posix_sysconf_SC_XOPEN_SHM			  _XOPEN_SHM
+#define __z_posix_sysconf_SC_XOPEN_STREAMS		  _XOPEN_STREAMS
+#define __z_posix_sysconf_SC_XOPEN_UNIX			  _XOPEN_UNIX
+#define __z_posix_sysconf_SC_XOPEN_UUCP			  _XOPEN_UUCP
+#define __z_posix_sysconf_SC_XOPEN_VERSION		  _XOPEN_VERSION
+#define __z_posix_sysconf_SC_CLK_TCK			  (100L)
+#define __z_posix_sysconf_SC_GETGR_R_SIZE_MAX		  (0L)
+#define __z_posix_sysconf_SC_GETPW_R_SIZE_MAX		  (0L)
+#define __z_posix_sysconf_SC_AIO_LISTIO_MAX		  AIO_LISTIO_MAX
+#define __z_posix_sysconf_SC_AIO_MAX			  AIO_MAX
+#define __z_posix_sysconf_SC_AIO_PRIO_DELTA_MAX		  AIO_PRIO_DELTA_MAX
+#define __z_posix_sysconf_SC_ARG_MAX			  ARG_MAX
+#define __z_posix_sysconf_SC_ATEXIT_MAX			  ATEXIT_MAX
+#define __z_posix_sysconf_SC_CHILD_MAX			  CHILD_MAX
+#define __z_posix_sysconf_SC_HOST_NAME_MAX		  HOST_NAME_MAX
+#define __z_posix_sysconf_SC_IOV_MAX			  IOV_MAX
+#define __z_posix_sysconf_SC_LOGIN_NAME_MAX		  LOGIN_NAME_MAX
+#define __z_posix_sysconf_SC_NGROUPS_MAX		  _POSIX_NGROUPS_MAX
+#define __z_posix_sysconf_SC_MQ_OPEN_MAX		  MQ_OPEN_MAX
+#define __z_posix_sysconf_SC_MQ_PRIO_MAX		  MQ_PRIO_MAX
+#define __z_posix_sysconf_SC_OPEN_MAX			  CONFIG_POSIX_MAX_FDS
+#define __z_posix_sysconf_SC_PAGE_SIZE			  PAGE_SIZE
+#define __z_posix_sysconf_SC_PAGESIZE			  PAGESIZE
+#define __z_posix_sysconf_SC_THREAD_DESTRUCTOR_ITERATIONS PTHREAD_DESTRUCTOR_ITERATIONS
+#define __z_posix_sysconf_SC_THREAD_KEYS_MAX		  PTHREAD_KEYS_MAX
+#define __z_posix_sysconf_SC_THREAD_STACK_MIN		  PTHREAD_STACK_MIN
+#define __z_posix_sysconf_SC_THREAD_THREADS_MAX		  PTHREAD_THREADS_MAX
+#define __z_posix_sysconf_SC_RTSIG_MAX			  RTSIG_MAX
+#define __z_posix_sysconf_SC_SEM_NSEMS_MAX		  SEM_NSEMS_MAX
+#define __z_posix_sysconf_SC_SEM_VALUE_MAX		  SEM_VALUE_MAX
+#define __z_posix_sysconf_SC_SIGQUEUE_MAX		  SIGQUEUE_MAX
+#define __z_posix_sysconf_SC_STREAM_MAX			  STREAM_MAX
+#define __z_posix_sysconf_SC_SYMLOOP_MAX		  SYMLOOP_MAX
+#define __z_posix_sysconf_SC_TIMER_MAX			  TIMER_MAX
+#define __z_posix_sysconf_SC_TTY_NAME_MAX		  TTY_NAME_MAX
+#define __z_posix_sysconf_SC_TZNAME_MAX			  TZNAME_MAX
+
+#define sysconf(x) (long)UTIL_EVAL(__z_posix_sysconf ## x)
```

Comment:
`CONCAT` in `util.h` might be better suited for this

```suggestion
#define sysconf(x) (long)CONCAT(__z_posix_sysconf, x)
```

https://github.com/zephyrproject-rtos/zephyr/blob/592647cf0b8adb713bdd98bc7e1d67986fda4d3b/include/zephyr/sys/util.h#L271-L283

The `UTIL_EVAL` is in the `util_internal.h`, which provides _'helper macros needed by sys/util.h'_